app/src/main/java/tech/ula/ui/FilesystemEditFragment.kt

```diff
@@ -83,30 +81,98 @@ class FilesystemEditFragment : Fragment() {
         }
     }
 
+    fun setupTextInputs() {
+        input_filesystem_name.setText(filesystem.name)
+        input_filesystem_username.setText(filesystem.defaultUsername)
+        input_filesystem_password.setText(filesystem.defaultPassword)
+        input_filesystem_vncpassword.setText(filesystem.defaultVncPassword)
+
+        if (editExisting) {
+            input_filesystem_username.isEnabled = false
+            input_filesystem_password.isEnabled = false
+            input_filesystem_vncpassword.isEnabled = false
+        }
+
+        input_filesystem_name.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+                filesystem.name = p0.toString()
+            }
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+        })
+
+        input_filesystem_username.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+                filesystem.defaultUsername = p0.toString()
+            }
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+        })
+
+        input_filesystem_password.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+                filesystem.defaultPassword = p0.toString()
+            }
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+        })
+
+        input_filesystem_vncpassword.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+                filesystem.defaultVncPassword = p0.toString()
+            }
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+        })
+    }
+
     private fun insertFilesystem(): Boolean {
         val navController = NavHostFragment.findNavController(this)
-        if (filesystem.name == "") input_filesystem_name.error = getString(R.string.error_filesystem_name)
-        if (filesystem.name == "") {
-            Toast.makeText(activityContext, R.string.error_empty_field, Toast.LENGTH_LONG).show()
+        if (!isFilesystemParametersCorrect()) {
```

Comment:
I've done this myself tons of places and still catch myself doing it now, but I think this reads better as something like:

`if (!filesystemParametersAreCorrect())`

To me that just matches more closely how I would actually want to say that conditional out loud to someone, and it makes it easier to think about. Personal preference thing though.